Orange-OpenSource/its-client
# java/iot3mobility: implement send and receive of mobility messages

---
**Features**;

IoT3 Mobility SDK:
    * Implement send and receive of CAM (+ receive of abstracted object RoadUser)
    * Implement send and receive of DENM (+ receive of abstracted object RoadHazard)
    * Implement send and receive of CPM (+ receive of abstracted objects RoadSensor and SensorObject)

Closes: #129 

---
**How to test**

1. Clone the its-client project on your IDE.
2. Ensure that you have Gradle to be able to build the java/iot3 modules (core, mobility and examples).
3. Find the ```Iot3MobilityExample``` class in the ```examples``` module, and set appropriate values for the following fields:
```private static final String EXAMPLE_HOST = "host";
    private static final String EXAMPLE_USERNAME = "username";
    private static final String EXAMPLE_PASSWORD = "password";
    private static final String EXAMPLE_UUID = "uuid";
    private static final String EXAMPLE_CONTEXT = "context";
    private static final String EXAMPLE_TELEMETRY_HOST = "telemetry_host";
```
4. Run ```Iot3MobilityExample```.

---
**Expected results:**

1. CAMs should be sent and received at 1 Hz, and visible in the logs.
2. DENMs should be sent and received every 10 seconds, and visible in the logs.
3. CPMs should be sent and received at 1 Hz, and visible in the logs.

You can also use the ITSdroid app (with a project code matching the EXAMPLE_CONTEXT value), to visualise the messages at the UTAC TEQMO.
